### PR TITLE
Add errors for polling

### PR DESCRIPTION
**Pull request by v1r3n**

### CODE REVIEW

File: sdk/worker/task_runner.go [hunk 1]
```diff
@@ -529,20 +534,11 @@ func (c *TaskRunner) GetBatchSizeForTask(taskName string) (batchSize int) {
...
-func sleep(taskName string, domain string, err error, sleepFor time.Duration) {
-	log.Trace(
-		"pausing worker",
-		", taskName: ", taskName,
-		", domain: ", domain,
-		", for: ", sleepFor.Milliseconds(), "ms",
-		", reason: ", err.Error(),
-	)
-	time.Sleep(sleepFor)
+	log.Trace(fmt.Errorf("no worker available for the task %s", taskName))
```
Comment (coderabhigupta) [inline sdk/worker/task_runner.go:542]: Do we want to log the domain also here?

File: sdk/worker/task_runner.go [hunk 2]
```diff
@@ -288,7 +296,10 @@ func (c *TaskRunner) executeAndUpdateTask(taskName string, task model.Task, exec
 	defer c.runningWorkerDone(taskName)
 	defer concurrency.HandlePanicError("execute_and_update_task " + string(task.TaskId) + ": " + string(task.Status))
 	taskResult := c.executeTask(&task, executeFunction)
-	c.updateTaskWithRetry(taskName, taskResult)
+	err := c.updateTaskWithRetry(taskName, taskResult)
+	if err != nil {
```
Comment (coderabhigupta) [inline sdk/worker/task_runner.go:300]: Should we call `pauseOnGenericError` here?

Comment (v1r3n) [reply sdk/worker/task_runner.go:300]: no, retry already does that.

File: sdk/worker/task_runner.go (hunk 1 above)
Comment (v1r3n) [reply sdk/worker/task_runner.go]: fixed.

**Pull request merged**